### PR TITLE
Bileworm Health Patch: Evolutionary Leap, Vileworms, Proximity Spawning Rules, Better Loot

### PR DESCRIPTION
**Pull request by tralezab**
### Bileworm Changes

- Bileworms only shoot twice
- Bileworms have half health
- Bileworms have a higher cooldown before reburrowing after shooting
- Bileworms now drop a bit of gold
- Bileworms no longer pierce armor

### Vileworms

- After 30 minutes past roundstart, all bileworms will evolve into Vileworms.
- Vileworms have the old stats of pre-nerf bileworms, plus occasionally an extra attack
- Same rewards

### Bileworm Fixes

- Discovered a bug that meant vileworms were breaking the spawning rules around not spawning near other mobs. This has been fixed!

#### Todo

- [ ] Add vileworm sprites

## Why It's Good For The Game

This is a more conscious nerf to bileworms, with feedback taken. Simply lowering a damage value isn't the way to fix bileworms, as loot was a problem, and DEFINITELY the spawning bug. I see a lot of complaints about getting "ganged up on" by bileworms, that should be resolved.

## Changelog

:cl:
add: After 30 minutes, Bileworms evolve into Vileworms, a more dangerous variant.
balance: Bileworms are easier
balance: Bileworms drop gold
fix: fixes bileworms spawning too close to other mobs, causing very, VERY tough fights
/:cl:

### CODE REVIEW

**Comment by VladinXXV** [comment]
I think the more difficult vileworms warrant _slightly_ better loot, even if it just comes down to a little extra gold or something. Silver from bileworms, gold from vileworms? Something like that.

**Comment by tralezab** [comment]
You still found a way to ask for more...

**Comment by VladinXXV** [comment]
> You still found a way to ask for more...

It's just a thought. Not complaining.

File: code/modules/mob/living/basic/lavaland/bileworm/_bileworm.dm [hunk 1]
```diff
@@ -17,6 +17,8 @@
 	guaranteed_butcher_results = list(
 		/obj/effect/gibspawner/generic = 1,
 		/obj/item/stack/sheet/animalhide/bileworm = 1,
+		/obj/item/stack/ore/gold,
+		/obj/item/stack/ore/gold,
```
Comment (Pickle-Coding) [inline code/modules/mob/living/basic/lavaland/bileworm/_bileworm.dm:21]: Was the duplicate entry intended?

Comment (tralezab) [reply code/modules/mob/living/basic/lavaland/bileworm/_bileworm.dm:21]: yeah, so it drops 2

**Comment by Wallemations** [comment]
I think if you added an option to the component for time-spread and made it a numerical value that could be really useful to have it so not every bile worm evolves at once

```
if(spread)
   evolve_time = rand(evolve_time * (1 - spread / 100), evolve_time)
```
On mobile so can't really give an exact code block but I hope this gets the idea across

**Comment by GuillaumePrata** [comment]
> Vileworms have the old stats of pre-nerf bileworms, plus occasionally an extra attack

It looks like Vileworms still lost their armor penetration.

But that just makes me happy, we give miners stupidly powerful armor and fauna just ignores it. Was always some backwards design were miner gear becomes oppressive against the station while not doing much against fauna.

File: code/__DEFINES/is_helpers.dm [hunk 2]
```diff
@@ -124,6 +124,11 @@ GLOBAL_LIST_INIT(turfs_openspace, typecacheof(list(
 /// returns whether or not the atom is either a basic mob OR simple animal
 #define isanimal_or_basicmob(A) (istype(A, /mob/living/simple_animal) || istype(A, /mob/living/basic))
 
+/// asteroid mobs, which are both simple and basic atm
+#define isminingpath(A) (ispath(A, /mob/living/simple_animal/hostile/asteroid) || ispath(A, /mob/living/basic/mining))
+
+#define ismining(A) (istype(A, /mob/living/simple_animal/hostile/asteroid) || istype(A, /mob/living/basic/mining))
```
Comment (Fikou) [inline code/__DEFINES/is_helpers.dm:130]: there's a lot of these checks for simple mining mobs in the codebase, would you mind looking at them to see which ones can be converted to use this?

File: code/modules/mob/living/basic/lavaland/bileworm/_bileworm.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/mob/living/basic/lavaland/bileworm/_bileworm.dm:21]: hahahahahaha

Comment (Fikou) [reply code/modules/mob/living/basic/lavaland/bileworm/_bileworm.dm:21]:
```suggestion
		/obj/item/stack/ore/gold = 2,
```

File: code/__DEFINES/is_helpers.dm (hunk 2 above)
Comment (tralezab) [reply code/__DEFINES/is_helpers.dm:130]: sure

**Comment by tralezab** [comment]
> > Vileworms have the old stats of pre-nerf bileworms, plus occasionally an extra attack
> 
> It looks like Vileworms still lost their armor penetration.
> 
> But that just makes me happy, we give miners stupidly powerful armor and fauna just ignores it. Was always some backwards design were miner gear becomes oppressive against the station while not doing much against fauna.

I decided not to give it back to Vileworms because if anything, the late game enemy should respect what armor you have obtained

**Comment by Jackal-boop** [comment]
maybe give vile worms a chance of dropping metal hydrogen, and a very rare chance of dropping zaukerite? would make them fun to hunt if I knew I could get the swag metals

**Comment by Fikou** [comment]
you didnt add the icons

**Comment by Shadowflame909** [comment]
I bet the syndicate is behind this vileworm buff

**Pull request merged**